client/state/reader/streams/reducer.js

```diff
@@ -16,50 +17,104 @@ import {
 	READER_STREAMS_UPDATES_RECEIVE,
 	READER_STREAMS_SELECT_NEXT_ITEM,
 	READER_STREAMS_SELECT_PREV_ITEM,
+	READER_STREAMS_SHOW_UPDATES,
 } from 'state/action-types';
-import { keyToString, keyForPost, keysAreEqual } from 'reader/post-key';
+import { keysAreEqual } from 'reader/post-key';
 
 /*
  * Contains a list of post-keys representing the items of a stream.
  */
 export const items = ( state = [], action ) => {
 	switch ( action.type ) {
 		case READER_STREAMS_PAGE_RECEIVE:
-			const { posts } = action.payload;
-			const postKeys = posts.map( keyForPost );
+			const { streamItems, gap } = action.payload;
 
-			const newState = uniqBy( [ ...state, ...postKeys ], keyToString );
+			let nextState;
+			if ( !! gap ) {
```

Comment:
why the `!!`?

Reply:
because its so exciting!!!!! I'll remove it :)